### PR TITLE
Drag-and-drop implementation

### PR DESCRIPTION
**Pull request by ahorobets**
Drag-and-drop functionality issues #380 #157

### CODE REVIEW

**Comment by jhoobergs** [comment]
@ArtemGorobets Sounds nice, can you explain exactly how I could test this and add some explanation about the new functions to the readme ?

**Comment by ahorobets** [comment]
@jhoobergs Sample App contains draggable view, you can try it

File: sample/src/main/res/layout/activity_base.xml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 <RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:tools="http://schemas.android.com/tools"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    tools:context=".BaseActivity"
-    android:background="#fff">
+                xmlns:tools="http://schemas.android.com/tools"
```
Comment (jhoobergs) [inline sample/src/main/res/layout/activity_base.xml]: Can you undo this formatting ?

File: sample/src/main/java/com/alamkanak/weekview/sample/BaseActivity.java [hunk 2]
```diff
@@ -211,4 +230,9 @@ public void onEmptyViewClicked(Calendar date) {
     public void onAddEventClicked(Calendar startTime, Calendar endTime) {
         Toast.makeText(this, "Add event clicked.", Toast.LENGTH_SHORT).show();
     }
+
+    @Override
+    public void onDrop(View view, Calendar date) {
+        Toast.makeText(this, "View dropped to " + getWeekView().getDateTimeInterpreter().interpretDate(date), Toast.LENGTH_SHORT).show();
```
Comment (jhoobergs) [inline sample/src/main/java/com/alamkanak/weekview/sample/BaseActivity.java]: Can you make it possible to set the text via a setter and a property ?

File: sample/src/main/java/com/alamkanak/weekview/sample/BaseActivity.java [hunk 3]
```diff
@@ -81,6 +91,15 @@ protected void onCreate(Bundle savedInstanceState) {
         setupDateTimeInterpreter(false);
     }
 
+    private final class DragTapListener implements View.OnLongClickListener {
+        @RequiresApi(api = Build.VERSION_CODES.HONEYCOMB)
+        @Override public boolean onLongClick(View v) {
+            ClipData               data          = ClipData.newPlainText("", "");
```
Comment (jhoobergs) [inline sample/src/main/java/com/alamkanak/weekview/sample/BaseActivity.java]: Can you fix this formatting ?

File: sample/src/main/res/layout/activity_base.xml [hunk 4]
```diff
@@ -26,4 +26,13 @@
...
         app:showHalfHours="false" />
 
+    <TextView
+        android:id="@+id/draggable_view"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:textColor="@android:color/white"
+        android:background="#ffff0000"
+        android:padding="3dp"
+        android:text="@string/drag_me"/>
```
Comment (jhoobergs) [inline sample/src/main/res/layout/activity_base.xml:36]: Can you make it possible to set this value with a setter or property ?

**Review by jhoobergs — COMMENTED** [review]
@ArtemGorobets I added a small comments. Code look good.

File: sample/src/main/res/layout/activity_base.xml (hunk 1 above)
Comment (ahorobets) [reply sample/src/main/res/layout/activity_base.xml]: done

File: sample/src/main/res/layout/activity_base.xml (hunk 4 above)
Comment (ahorobets) [reply sample/src/main/res/layout/activity_base.xml:36]: this view appears in sample app, not in library, i see no reason to make setter or property in this case

File: sample/src/main/java/com/alamkanak/weekview/sample/BaseActivity.java (hunk 3 above)
Comment (ahorobets) [reply sample/src/main/java/com/alamkanak/weekview/sample/BaseActivity.java]: done

File: sample/src/main/java/com/alamkanak/weekview/sample/BaseActivity.java (hunk 2 above)
Comment (ahorobets) [reply sample/src/main/java/com/alamkanak/weekview/sample/BaseActivity.java]: this info message appears in sample app, not in library, it was made in ordering to existing code styling

File: sample/src/main/res/layout/activity_base.xml (hunk 4 above)
Comment (jhoobergs) [reply sample/src/main/res/layout/activity_base.xml:36]: Yeah, you're right. Seems I shouldn't have reviewed it so late at night.

**Comment by jhoobergs** [comment]
@ArtemGorobets It seems alright, I will test it tonight, installing the latest Android build tools takes a lot of time. Just a small question because I'm not familiar with the drag and drop listener: 'Is the drag always enabled in the weekview with this implementation ?'

**Comment by ahorobets** [comment]
@jhoobergs It's just possibility to recognize position (date/time) where some view has been dropped (like a click on empty space). So this ability is useful when you have for example list of tasks and you need to plan your timetable, in this case drag-n-drop seems like a good solution - you can drag each item of listview on weekview for creating new events.

**Comment by jhoobergs** [comment]
@ArtemGorobets Everything is fine, except the fact that I want it to be possible to disable the dragListener at the weekview. When you, for example, have another view with a draglistener at the same page, it can come in handy to disable or enable the draglistener of the weekview. If you do that, i will gladly merge this and make a new release.

Something else: maybe we can add a 'dragable' property to WeekViewItems ? I think that might even be better. This is not something for this PR, but i think it would be a great other feature.

**Comment by ahorobets** [comment]
@jhoobergs WeekView based on canvas drawing, so there is no simple way to drag items like regular views

**Comment by jhoobergs** [comment]
@ArtemGorobets Ah, my bad. Can still fix my previous question ? 

> I want it to be possible to disable the dragListener at the weekview. When you, for example, have another view with a draglistener at the same page, it can come in handy to disable or enable the draglistener of the weekview.

**Review by kingnightdrifter — COMMENTED** [review]
I think in this changing you add functionality for the textview drag and drop but people asking for event drag and drop badly

**Pull request closed without merging**